h3_postgis/sql/install/40-rasters.sql

```diff
@@ -0,0 +1,653 @@
+/*
+ * Copyright 2022 Bytes & Brains
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+--| # Raster processing functions
+
+-- Get nodata value for ST_Clip function
+-- ST_Clip sets nodata pixel values to minimum value by default, but it won't
+-- set band nodata value in this case, which we need later for filtering dumped
+-- values.
+CREATE OR REPLACE FUNCTION __h3_raster_band_nodata(
+    rast raster,
+    nband integer)
+RETURNS double precision
+AS $$
+    SELECT coalesce(
+        ST_BandNoDataValue(rast, nband),
+        ST_MinPossibleValue(ST_BandPixelType(rast, nband)));
+$$ LANGUAGE SQL IMMUTABLE STRICT PARALLEL SAFE;
+
+CREATE OR REPLACE FUNCTION __h3_raster_to_polygon(
+    rast raster,
+    nband integer)
+RETURNS geometry
+AS $$
+    SELECT ST_MinConvexHull(rast, nband);
+$$ LANGUAGE SQL IMMUTABLE STRICT PARALLEL SAFE;
+
+-- Area of a single pixel in meters
+CREATE OR REPLACE FUNCTION __h3_raster_pixel_area(rast raster)
+RETURNS double precision
+AS $$
+    SELECT ST_Area(ST_Transform(ST_PixelAsPolygon(rast, 1, 1), 4326)::geography);
```

Comment:
I think so, but maybe we can instead use the area of a pixel at `ST_Centroid(ST_MinConvexHull(poly))`?
The difference should be insignificant for a case where all pixels have data, but it would be a bit more precise in case when the pixels are clustered near the edge of the raster (e.g. all "build up" pixels are in the south).

Reply:
Replaced with `__h3_raster_polygon_pixel_area`, area of pixel at `ST_Centroid(ST_MinConvexHull(rast))`.